src/main/java/paymelah/logic/parser/FindCommandParser.java

```diff
@@ -18,12 +36,96 @@ public class FindCommandParser implements Parser<FindCommand> {
      * @throws ParseException if the user input does not conform the expected format
      */
     public FindCommand parse(String args) throws ParseException {
-        try {
-            return new FindCommand(prepareNameContainsKeywordsPredicate(args));
-        } catch (ParseException e) {
-            throw new ParseException(
-                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindCommand.MESSAGE_USAGE));
+        ArgumentMultimap argMultimap =
+                ArgumentTokenizer.tokenize(args, PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS,
+                        PREFIX_TAG, PREFIX_DESCRIPTION, PREFIX_MONEY, PREFIX_DATE, PREFIX_TIME);
+
+        PersonDescriptor personDescriptor = new PersonDescriptor();
+        if (argMultimap.getValue(PREFIX_NAME).isPresent()) {
+            personDescriptor.setName(ParserUtil.parseName(argMultimap.getValue(PREFIX_NAME).get()));
+        }
+        if (argMultimap.getValue(PREFIX_PHONE).isPresent()) {
+            personDescriptor.setPhone(ParserUtil.parsePhone(argMultimap.getValue(PREFIX_PHONE).get()));
+        }
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            personDescriptor.setEmail(ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get()));
+        }
+        if (argMultimap.getValue(PREFIX_ADDRESS).isPresent()) {
+            personDescriptor.setAddress(ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get()));
+        }
+        parseTagsForFind(argMultimap.getAllValues(PREFIX_TAG)).ifPresent(personDescriptor::setTags);
+        parseDescriptionsForFind(argMultimap.getAllValues(PREFIX_DESCRIPTION))
+                .ifPresent(personDescriptor::setDescriptions);
+        parseMoniesForFind(argMultimap.getAllValues(PREFIX_MONEY)).ifPresent(personDescriptor::setMonies);
+        parseDatesForFind(argMultimap.getAllValues(PREFIX_DATE)).ifPresent(personDescriptor::setDates);
+        parseTimesForFind(argMultimap.getAllValues(PREFIX_TIME)).ifPresent(personDescriptor::setTimes);
```

Comment:
What's the rationale behind some fields using `getValue` and others using `getAllValue`?

Reply:
People can't have multiple names or phone numbers, but they can have multiple tags/debts